data_subscriber/daac_data_subscriber.py

```
@@ -331,101 +329,57 @@ def update_granule_index(es_spatial_conn, granule):
-    request.install_opener(opener)
 
-    return username, password
+def _get_tokens(edl: str, username: str, password: str) -> list:
+    token_list_url = f"https://{edl}/api/users/tokens"
 
+    list_response = requests.get(token_list_url, auth=HTTPBasicAuth(username, password))
+    list_content = json.loads(list_response.content)
 
-@contextmanager
-def token_ctx(edl):
+    return list_content
 
-    token_dict = _get_token(edl)
-    try:
-        yield token_dict
-    finally:
-        _delete_token(edl, token_dict)
 
+def _process_token_list(token_list: list, edl: str, username: str, password: str) -> None:
+    for token_dict in token_list:
+        now = datetime.utcnow().date()
+        expiration_date = datetime.strptime(token_dict['expiration_date'], "%m/%d/%Y").date()
 
-def _get_token(edl: str) -> dict:
-    token_create_url = f"https://{edl}/api/users/token"
-    token_list_url = f"https://{edl}/api/users/tokens"
-    username, _, password = netrc.netrc().authenticators(edl)
+        if expiration_date <= now:
+            _delete_token(edl, username, password, token_dict['access_token'])
 
-    list_response = requests.get(token_list_url, auth=HTTPBasicAuth(username, password))
-    list_content = json.loads(list_response.content)
 
-    if not list_content:
-        create_response = requests.post(token_create_url, auth=HTTPBasicAuth(username, password))
-        response_content = json.loads(create_response.content)
+def _create_token(edl: str, username: str, password: str):
+    token_create_url = f"https://{edl}/api/users/token"
+
+    create_response = requests.post(token_create_url, auth=HTTPBasicAuth(username, password))
+    response_content = json.loads(create_response.content)
 
-        if "error" in response_content.keys():
-            logging.warning("Failed to acquire CMR token")
-            raise Exception(response_content['error'])
+    if "error" in response_content.keys():
+        logging.warning("Failed to acquire CMR token")
```

because this condition results in an exception being thrown, consider not logging at the `warning` level. you can simply let the exception bubble up and the error be expressed in the exception message.